### ISSUE TITLE
Toggling Show MS2 Scans causes crash

### ISSUE THREAD

**Issue by PMSeitzer** [eugenemel/maven#229]
To reproduce:

load large dataset with all adducts.

Toggle right click

 crash with segmentation fault:
```
	computeEICs() Done.

 EicWidget::replot(PeakGroup* group ) group= 0x0

EicWidget::findPlotBounds() _minX= 6.40655 minutes, _maxX= 9.47374  minutes

addMS2Events()  found= 0

MainWindow::setPeakGroup(PeakGroup) 0x7fe047de2788

EicWidget::setPeakGroup(PeakGroup* group)  0x7fe047de2788

 EicWidget::recompute()

 EicWidget::computeEICs()

	computeEICs() pullEics().. msec= 79

	groupPeaks() msec= 96

	computeEICs() Done.

 EicWidget::replot(PeakGroup* group ) group= 0x7fe047de2788

EicWidget::findPlotBounds() _minX= 6.86075 minutes, _maxX= 9.42526  minutes

addMS2Events()  found= 78

Isotope Calculation:  true false false false

MassCalculator::getComposition(Adduct*)

SpectraWidget::setCurrentFragment(fragment)

SpectraWidget::setCurrentScan(scan)

overlayCompound() LPE(18:1/0:0) [M+H]+ #theory=16	 #obs=428	 #matched=0

SpectraWidget::setScan(scan,min,max) : 1330

SpectraWidget::setCurrentScan(scan)

MassCalcWidget::setPeakGroup()

MainWindow::showPeakInfo(peak)

SpectraWidget::setScan(peak) 

SpectraWidget::setCurrentScan(scan)

Isotope Calculation:  true false false false

MassCalculator::getComposition(Adduct*)

EicWidget::contextMenuEvent(QContextMenuEvent * event)  QGraphicsItem(0)

EicWidget::contextMenuEvent(QContextMenuEvent * event)  QGraphicsItem(0x7fe197b8a5e0, pos=0,0, z=14)

EicWidget::contextMenuEvent(QContextMenuEvent * event)  QGraphicsItem(0)

 EicWidget::replot(PeakGroup* group ) group= 0x7fe229819190

EicWidget::findPlotBounds() _minX= 6.86075 minutes, _maxX= 9.42526  minutes

QGraphicsItem::ungrabMouse: not a mouse grabber

Segmentation fault: 11

logout

Saving session...

...copying shared history...

...saving history...truncating history files...

...completed.

Deleting expired sessions...17 completed.



[Process completed]
```

**Comment by PMSeitzer** [comment]
toggling the `Group Peaks` also causes the bug, which eventually leads to a `segmentation fault`.

`X0144-M004A` is the dataset

**Comment by PMSeitzer** [comment]
May be associated with #255 #235 and #249 (peak grouping with Savitzky-Golay smoothing could be the fundamental issue).

**Comment by PMSeitzer** [comment]
This has not been observed recently, thought to be fixed perhaps by #277
